### PR TITLE
Add [extension/sigv4authextension] to ADOT Collector (PR)

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:**
This PR adds the `sigv4authextension` to the ADOT Collector. It also adds the `prometheusremotewrite` exporter as a factory, as they together will eventually replace the `awsprometheusremotewrite` exporter

**Link to tracking Issue:**
#1065 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by erichsueh3** [comment]
A few notes:
1. Not sure why a commit for `test_canary.yml` was included, maybe it was pushed to the main branch of `open-o11y`?
2. In `go.mod`, since the sigv4 extension didn't exist in 0.46, how do we deal with this? Do we need to wait until 0.18 of the Collector is prepped before we can add it in as 0.47?
3. Unsure if the PRW exporter + sigv4 extension changes needs to be added to `pkg/lambdacomponents`? I would think the answer is yes, and if it is can be quickly added but have not done it as of yet.
4. The `config/` and `examples/` yaml files will be updated when the AWS PRW exporter is eventually taken out.
5. The AWS PRW exporter will be removed in release 0.19(?); as for a deprecation notice, not sure where I would add it, or if I even need to, since once 0.18 comes out, it will use 0.47 of the AWS PRW exporter that has the notice there already, but correct me if I'm wrong.

File: go.mod [hunk 1]
```diff
@@ -12,12 +12,14 @@ require (
...
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter v0.46.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter v0.46.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter v0.46.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter v0.46.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/awsproxy v0.46.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension v0.46.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/ecsobserver v0.46.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/pprofextension v0.46.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/extension/sigv4authextension v0.47.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/processor/attributesprocessor v0.46.0
```
Comment (bryan-aguilar) [inline go.mod:23]: v0.47.0 upgrade should becoming today but as a thought exercise would this cause failure for the user if we shipped a version mismatch like this? Should we allow PRs with a collector version mis match for components?

File: README.md [hunk 2]
```diff
@@ -25,18 +25,20 @@ This table represents the supported components of the ADOT Collector. The highli
...
-| jaegerreceiver                  | spanprocessor                 | otlphttpexporter                   |                        |
+| jaegerreceiver                  | spanprocessor                 | otlphttpexporter                   | `sigv4authextension`   |
 | `awscontainerinsightreceiver`   | filterprocessor               | prometheusexporter                 |                        |
 |                                 | resourcedetectionprocessor    | datadogexporter                    |                        |
 |                                 | `metricsgenerationprocessor`  | dynatraceexporter                  |                        |
 |                                 | cumulativetodeltaprocessor    | sapmexporter                       |                        |
 |                                 | deltatorateprocessor          | signalfxexporter                   |                        |
 |                                 |                               | logzioexporter                     |                        |
+|                                 |                               | prometheusremotewriteexporter      |                        |
```
Comment (bryan-aguilar) [inline README.md:39]: A link to a corresponding in the `aws-otel.github.io` repository needs to be linked here so that the tables match.

**Comment by Aneurysm9** [comment]
> Are test cases going to be added to the `aws-otel-test-framework` to test these components?

We should have the same tests that are run with the `awsprometheusremotewrite` exporter configured to run with the `prometheusremotewrite` exporter and `sigv4auth` extension.

**Comment by erichsueh3** [comment]
> > Are test cases going to be added to the `aws-otel-test-framework` to test these components?
> 
> We should have the same tests that are run with the `awsprometheusremotewrite` exporter configured to run with the `prometheusremotewrite` exporter and `sigv4auth` extension.

Is this something we want to change now, or change in the future once we remove the `awsprometheusremotewrite` exporter?

**Comment by bryan-aguilar** [comment]
> Is this something we want to change now, or change in the future once we remove the awsprometheusremotewrite exporter?
A new component should not be added without included tests in my opinion.

File: README.md (hunk 2 above)
Comment (erichsueh3) [reply README.md:39]: Filed https://github.com/aws-otel/aws-otel.github.io/pull/270 in `aws-otel/aws-otel.github.io`

File: pkg/lambdacomponents/defaults.go [hunk 3]
```diff
@@ -36,6 +38,13 @@ func Components() (
 ) {
 	var errs error
 
+	extensions, err := component.MakeExtensionFactoryMap(
+		sigv4authextension.NewFactory(),
```
Comment (bryan-aguilar) [inline pkg/lambdacomponents/defaults.go:42]: This extension factory  map needs to be added to the returned `factories` struct at the end of the function.

**Review by bryan-aguilar — CHANGES_REQUESTED** [review]

File: pkg/lambdacomponents/defaults.go (hunk 3 above)
Comment (erichsueh3) [reply pkg/lambdacomponents/defaults.go:42]: Fixede

**Review by bryan-aguilar — APPROVED** [review]

**Review by vasireddy99 — APPROVED** [review]
LGTM!

**Pull request merged**